project/leeshengcian/README.md

```diff
@@ -0,0 +1,74 @@
+# Visualization-of-Dijkstra-Algorithm
+This Proposal shows the work flow and plans for the 2023 Autumn NYCU NSD term project
+
+## Basic Information
+
+GitHub Link : [https://github.com/leeshengcian/Visualization-of-Dijkstra-Algorithm/tree/main]
+
+Dijkstra's algorithm is an algorithm for finding the shortest paths between 
```

Comment:
How do you measure/represent the distance between nodes?

Reply:
The graph data will be stored in adjacency list format, which the length of edges represents the distance between nodes

Reply:
How do you measure the length of edges?

Reply:
In my case, in graph theory we don't have the concept of measurement, which we won't say the distance between A_node to B_node is 5 meters. Instead, the numeric of edge length represent how many multiples compared with "unit length"

For instance, below is a sample input of graph data, 
first row represent the number of nodes(eg:5)
As for second to sixth row, each represent the adjacency list of each node
Take second row(0, 2, 7, -1, -1) for example, 
0 is a default value to determine whether this is a source node
2 means: the edge length from source node to node_#1 is 2 unit length
7 means: the edge length from source node to node_#2 is 7 unit length
-1 means: there is no edge connected between source node and node_#3 (and node_#4)
<img width="128" alt="image" src="https://github.com/yungyuc/nsdhw_23au/assets/145427848/73d87c72-42de-49c9-a53c-ac85a0797cc6">

That's my initial idea about this work, maybe we can discuss a better solution for measurement of edge length at tomorrow meetup

Reply:
If all edges have the same length, there will be no distinguish between long and short edges.  I think it may be productive for you to come up with a proposal for measuring the edge length.  Later if we meet we can move forward something else.

Reply:
It is not practical to use a GUI to manually assign length for a large amount of data, e.g., 10,000 nodes.  You will need a way to assign or determine length programmatically.

Reply:
Then I would assign a floating point value to the "unit length" and each edge has length of multiple of that floating point value
This is a rough idea about determine length, we should discuss more